snowplow_tracker/events/screen_view.py

```diff
@@ -0,0 +1,227 @@
+# """
+#     screen_view.py
+
+#     Copyright (c) 2013-2023 Snowplow Analytics Ltd. All rights reserved.
+
+#     This program is licensed to you under the Apache License Version 2.0,
+#     and you may not use this file except in compliance with the Apache License
+#     Version 2.0. You may obtain a copy of the Apache License Version 2.0 at
+#     http://www.apache.org/licenses/LICENSE-2.0.
+
+#     Unless required by applicable law or agreed to in writing,
+#     software distributed under the Apache License Version 2.0 is distributed on
+#     an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+#     express or implied. See the Apache License Version 2.0 for the specific
+#     language governing permissions and limitations there under.
+# """
+
+from typing import Optional, List
+from snowplow_tracker.typing import JsonEncoderFunction
+from snowplow_tracker.events.event import Event
+from snowplow_tracker import SelfDescribingJson
+from snowplow_tracker.constants import (
+    UNSTRUCT_EVENT_SCHEMA,
+    CONTEXT_SCHEMA,
+    MOBILE_SCHEMA_PATH,
+    SCHEMA_TAG,
+)
+from snowplow_tracker import payload
+from snowplow_tracker import subject as _subject
+
+
+class ScreenView(Event):
+    """
+    Constructs a ScreenView event object.
+
+    When tracked, generates a SelfDescribing event (event type "ue").
+    """
+
+    def __init__(
+        self,
+        event_subject: Optional[_subject.Subject] = None,
+        context: Optional[List[SelfDescribingJson]] = None,
+        tstamp: Optional[float] = None,
+        id_: Optional[str] = None,
```

Comment:
The name and the id are required properties in the schema so should not be `Optional`. They shouldn't have a default value and should be first in the list of parameters.